Content.Server/Chemistry/Metabolism/DefaultDrink.cs

```diff
@@ -23,9 +23,17 @@ public class DefaultDrink : IMetabolizable
         public float HydrationFactor { get; set; } = 30.0f;
 
         //Remove reagent at set rate, satiate thirst if a ThirstComponent can be found
-        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime)
+        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime, ReagentUnit availableReagent)
         {
+            // how much reagant should we metabolize
             var metabolismAmount = MetabolismRate * tickTime;
+
+            // is that much reagant actually available?
+            if (availableReagent < metabolismAmount)
+            {
+                metabolismAmount = availableReagent;
+            }
```

Comment:
as said before this check should be handled by liverbehavior after this  function is called, won't comment on the other usages but same goes there

Reply:
The issue of wheter the check should be here aside, just the fact that the code for the "is that much reagant actually available?" test appears repeated in every single Metabolizable Classes is bad practice, and should probably be done differently.